lib/src/spotify_base.dart

```diff
@@ -83,6 +81,18 @@ abstract class SpotifyApiBase {
         );
       }
 
+      if (callBack != null) {
+        return oauth2.Client(
+          oauthCredentials,
+          identifier: credentials.clientId,
+          onCredentialsRefreshed: (oauth2.Credentials cred) {
+            SpotifyApiCredentials newCredentials = SpotifyApiCredentials(credentials.clientId, credentials.clientSecret,
+                accessToken: cred.accessToken, expiration: cred.expiration, refreshToken: cred.refreshToken, scopes: cred.scopes);
+            callBack(newCredentials);
+          },
+          secret: credentials.clientSecret,
+        );
+  }
       return oauth2.Client(
         oauthCredentials,
         identifier: credentials.clientId,
```

Comment:
Another little thing, maybe this is better instead of the if
```dart
        return oauth2.Client(
          oauthCredentials,
          identifier: credentials.clientId,
          onCredentialsRefreshed: callBack == null ? null : (oauth2.Credentials cred) {
            SpotifyApiCredentials newCredentials = SpotifyApiCredentials(credentials.clientId, credentials.clientSecret,
                accessToken: cred.accessToken, expiration: cred.expiration, refreshToken: cred.refreshToken, scopes: cred.scopes);
            callBack(newCredentials);
          },
          secret: credentials.clientSecret,
        );
```